src/server/middleware/log/log.go

```diff
@@ -40,6 +45,61 @@ func Middleware() func(http.Handler) http.Handler {
 			r = r.WithContext(ctx)
 		}
 
-		next.ServeHTTP(w, r)
+		e := &commonevent.Metadata{
+			Ctx:           r.Context(),
+			Username:      "unknown",
+			RequestMethod: r.Method,
+			RequestURL:    r.URL.String(),
+		}
+		if matched, resName := e.PreCheck(); matched {
+			lib.NopCloseRequest(r)
```

Comment:
If we add this line, each request(almost each) will become a no-op close, and for certain requests, such as PUT manifest, it will be called twice.

Does that make sense?

Reply:
The `lib.NopCloseRequest(r)` is called after the PreCheck() is matched, the audit log middleware just handles the request of `/api/v2.0/*`, the PUT manifest is the request to `/v2/`